File: data/translated_projects/commons-fileupload/src/main/org/apache/commons/fileupload/util/LimitedInputStream.py
from __future__ import annotations
from abc import ABC
from io import StringIO
import io
from io import BytesIO
import typing
from typing import *
import os
from src.main.org.apache.commons.fileupload.util.Closeable import *
import java # type: ignore
from src.main.org.apache.commons.fileupload.java_handler import *
class LimitedInputStream(ABC, metaclass=StaticFieldRedirector):
    def close(self) -> None:

        idMapPyToJ = self.javaObj.pyToJ()
        
        idMapJToPy = dict()
        try:
            self.javaObj.close()
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            self.javaObj.jToPy(idMapJToPy)
            
    def isClosed(self) -> bool:

        idMapPyToJ = self.javaObj.pyToJ()
        
        idMapJToPy = dict()
        try:
            val = JavaHandler.mapping(self.javaObj.isClosed(), id_map=idMapJToPy)
            return val
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            self.javaObj.jToPy(idMapJToPy)
            
    def read1(self, b: typing.List[int], off: int, len_: int) -> int:

        idMapPyToJ = self.javaObj.pyToJ()
        translatedArg0 = b
        translatedArg1 = JavaHandler.valueToObject(off, "int", idMapPyToJ)
        translatedArg2 = JavaHandler.valueToObject(len_, "int", idMapPyToJ)
        idMapJToPy = dict()
        try:
            val = JavaHandler.mapping(self.javaObj.read1(translatedArg0, translatedArg1, translatedArg2), id_map=idMapJToPy)
            return val
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            self.javaObj.jToPy(idMapJToPy)
            JavaHandler.mapping(translatedArg0, idMapJToPy, b)
            JavaHandler.mapping(translatedArg1, idMapJToPy, off)
            JavaHandler.mapping(translatedArg2, idMapJToPy, len_)
    def read0(self) -> int:

        idMapPyToJ = self.javaObj.pyToJ()
        
        idMapJToPy = dict()
        try:
            val = JavaHandler.mapping(self.javaObj.read0(), id_map=idMapJToPy)
            return val
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            self.javaObj.jToPy(idMapJToPy)
            
    def __init__(self, inputStream: typing.Union[io.BytesIO, io.StringIO, io.BufferedReader], pSizeMax: int) -> None:

        idMapPyToJ = JavaHandler.valueToObject(dict(), "Map")
        translatedArg0 = JavaHandler.valueToObject(inputStream, "InputStream", idMapPyToJ)
        translatedArg1 = JavaHandler.valueToObject(pSizeMax, "long", idMapPyToJ)
        idMapJToPy = dict()
        try:
            self.javaObj = LimitedInputStream.javaClz(translatedArg0, translatedArg1)
            self.javaObj.setPythonObject(self)
            self.javaObj.jToPy(idMapJToPy)
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            JavaHandler.mapping(translatedArg0, idMapJToPy, inputStream)
            JavaHandler.mapping(translatedArg1, idMapJToPy, pSizeMax)
    def __checkLimit(self) -> None:

        idMapPyToJ = self.javaObj.pyToJ()
        
        idMapJToPy = dict()
        try:
            self.javaObj.checkLimit()
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            self.javaObj.jToPy(idMapJToPy)
            
    def _raiseError(self, pSizeMax: int, pCount: int) -> None:
        pass


    javaClz = java.type("org.apache.commons.fileupload.util.LimitedInputStream")
    def __getattr__(self, name):
        try:
            return object.__getattribute__(self, name)
        except AttributeError:
            pass
        return JavaHandler.mapping(getattr(self.javaClz, StaticFieldRedirector.unmangle_name(name)))

File: logs/glue/commons-fileupload/src/main/java/org/apache/commons/fileupload/util/LimitedInputStream.java
package org.apache.commons.fileupload.util;

import java.io.FilterInputStream;
import java.io.IOException;
import java.io.InputStream;
import org.apache.commons.fileupload.ContextInitializer;
import org.apache.commons.fileupload.ExceptionHandler;
import org.apache.commons.fileupload.IntegrationUtils;
import org.graalvm.polyglot.PolyglotException;
import org.graalvm.polyglot.Value;

public abstract class LimitedInputStream extends FilterInputStream implements Closeable {
  private transient Value obj =
      IntegrationUtils.createDefaultPythonObject(
          ContextInitializer.getPythonClass(
              "main/org/apache/commons/fileupload/util/LimitedInputStream.py",
              "LimitedInputStream"));
  public final long sizeMax;
  public long count;
  public boolean closed;

  public Value getPythonObject() {
    return obj;
  }

  public void setPythonObject(Value obj) {
    this.obj = obj;
  }

  @Override
  public void close() throws IOException {

    try {

      closed = true;
      super.close();

    } catch (Throwable ExceptionObjectForCaching) {
      ExceptionHandler.ERR = ExceptionObjectForCaching;
      throw ExceptionObjectForCaching;
    }
  }

  @Override
  public boolean isClosed() throws IOException {

    try {

      return closed;

    } catch (Throwable ExceptionObjectForCaching) {
      ExceptionHandler.ERR = ExceptionObjectForCaching;
      throw ExceptionObjectForCaching;
    }
  }

  public int read1(byte[] b, int off, int len) throws IOException {

    try {

      int res = super.read(b, off, len);
      if (res > 0) {
        count += res;
        checkLimit();
      }
      return res;

    } catch (Throwable ExceptionObjectForCaching) {
      ExceptionHandler.ERR = ExceptionObjectForCaching;
      throw ExceptionObjectForCaching;
    }
  }

  public int read0() throws IOException {

    try {

      int res = super.read();
      if (res != -1) {
        count++;
        checkLimit();
      }
      return res;

    } catch (Throwable ExceptionObjectForCaching) {
      ExceptionHandler.ERR = ExceptionObjectForCaching;
      throw ExceptionObjectForCaching;
    }
  }

  public LimitedInputStream(InputStream inputStream, long pSizeMax) {

    super(inputStream);
    sizeMax = pSizeMax;
  }

  public void checkLimit() throws IOException {

    try {

      if (count > sizeMax) {
        raiseError(sizeMax, count);
      }

    } catch (Throwable ExceptionObjectForCaching) {
      ExceptionHandler.ERR = ExceptionObjectForCaching;
      throw ExceptionObjectForCaching;
    }
  }

  protected abstract void raiseError(long pSizeMax, long pCount) throws IOException;

  public java.util.Map pyToJ() {
    java.util.Map idMap = new java.util.HashMap();
    return pyToJ(idMap);
  }

  public java.util.Map pyToJ(java.util.Map idMap) {

    this.count =
        (long)
            IntegrationUtils.valueToObject(
                this.obj.getMember("_LimitedInputStream__count"), "long", idMap, count);
    this.closed =
        (boolean)
            IntegrationUtils.valueToObject(
                this.obj.getMember("_LimitedInputStream__closed"), "boolean", idMap, closed);
    return idMap;
  }

  public Value jToPy() {
    Value idMap = IntegrationUtils.mapToPython(new java.util.HashMap());
    return jToPy(idMap);
  }

  public Value jToPy(Value idMap) {

    this.obj.invokeMember("__setattr__", "javaObj", this);
    this.obj.invokeMember(
        "__setattr__",
        "_LimitedInputStream__sizeMax",
        IntegrationUtils.mapToPython(
            sizeMax, idMap, this.obj.getMember("_LimitedInputStream__sizeMax")));
    this.obj.invokeMember(
        "__setattr__",
        "_LimitedInputStream__count",
        IntegrationUtils.mapToPython(
            count, idMap, this.obj.getMember("_LimitedInputStream__count")));
    this.obj.invokeMember(
        "__setattr__",
        "_LimitedInputStream__closed",
        IntegrationUtils.mapToPython(
            closed, idMap, this.obj.getMember("_LimitedInputStream__closed")));
    return idMap;
  }
}
